### PR TITLE
Expose `client_async_tls_with_connector_and_config`

### PR DESCRIPTION
**Pull request by lambdalisue**
Hi. First of all, thanks for your great library.

I needed to specify `stream`, `connector`, and `config` to make HTTP proxy and TLS enabled configurable websocket connection so I exposed that private function.

Please review it.

### CODE REVIEW

File: src/async_std.rs [hunk 1]
```diff
@@ -146,15 +146,21 @@ pub(crate) mod dummy_tls {
 }
 
 #[cfg(not(any(feature = "async-tls", feature = "async-native-tls")))]
-use self::dummy_tls::{client_async_tls_with_connector_and_config, AutoStream};
+pub use self::dummy_tls::client_async_tls_with_connector_and_config;
```
Comment (sdroege) [inline src/async_std.rs:152]: Can you do the same in the tokio module, and also make sure that this re-exported function is documented in all configurations?

**Comment by lambdalisue** [comment]
I built docs with the following

```
cargo doc --all-features
```

CleanShot 2021-06-29 at 17 12 06
CleanShot 2021-06-29 at 17 12 29
CleanShot 2021-06-29 at 17 12 49

File: src/async_std.rs (hunk 1 above)
Comment (sdroege) [reply src/async_std.rs:152]: If only the dummy one is enabled then it won't have documentation. You'll have to copy the docs to every variant of the function :)

Comment (lambdalisue) [reply src/async_std.rs:152]: Aha, I see.

Comment (lambdalisue) [reply src/async_std.rs:152]: Done. Confirmed by

```
git grep -B 'fn client_async_tls_with_connector_and_config'
```

```
src/async_std.rs-    /// Creates a WebSocket handshake from a request and a stream,
src/async_std.rs-    /// upgrading the stream to TLS if required and using the given
src/async_std.rs-    /// connector and WebSocket configuration.
src/async_std.rs:    pub async fn client_async_tls_with_connector_and_config<R, S>(
--
src/async_std.rs-    /// Creates a WebSocket handshake from a request and a stream,
src/async_std.rs-    /// upgrading the stream to TLS if required and using the given
src/async_std.rs-    /// connector and WebSocket configuration.
src/async_std.rs:    pub(crate) async fn client_async_tls_with_connector_and_config<R, S>(
--
src/async_tls.rs-/// Creates a WebSocket handshake from a request and a stream,
src/async_tls.rs-/// upgrading the stream to TLS if required and using the given
src/async_tls.rs-/// connector and WebSocket configuration.
src/async_tls.rs:pub async fn client_async_tls_with_connector_and_config<R, S>(
--
src/tokio/async_tls.rs-/// Creates a WebSocket handshake from a request and a stream,
src/tokio/async_tls.rs-/// upgrading the stream to TLS if required and using the given
src/tokio/async_tls.rs-/// connector and WebSocket configuration.
src/tokio/async_tls.rs:pub async fn client_async_tls_with_connector_and_config<R, S>(
--
src/tokio/dummy_tls.rs-/// Creates a WebSocket handshake from a request and a stream,
src/tokio/dummy_tls.rs-/// upgrading the stream to TLS if required and using the given
src/tokio/dummy_tls.rs-/// connector and WebSocket configuration.
src/tokio/dummy_tls.rs:pub(crate) async fn client_async_tls_with_connector_and_config<R, S>(
--
src/tokio/native_tls.rs-/// Creates a WebSocket handshake from a request and a stream,
src/tokio/native_tls.rs-/// upgrading the stream to TLS if required and using the given
src/tokio/native_tls.rs-/// connector and WebSocket configuration.
src/tokio/native_tls.rs:pub async fn client_async_tls_with_connector_and_config<R, S>(
--
src/tokio/openssl.rs-/// Creates a WebSocket handshake from a request and a stream,
src/tokio/openssl.rs-/// upgrading the stream to TLS if required and using the given
src/tokio/openssl.rs-/// connector and WebSocket configuration.
src/tokio/openssl.rs:pub async fn client_async_tls_with_connector_and_config<R, S>(
--
src/tokio/rustls.rs-/// Creates a WebSocket handshake from a request and a stream,
src/tokio/rustls.rs-/// upgrading the stream to TLS if required and using the given
src/tokio/rustls.rs-/// connector and WebSocket configuration.
src/tokio/rustls.rs:pub async fn client_async_tls_with_connector_and_config<R, S>(
```

**Comment by lambdalisue** [comment]
Shall I squash commits?

**Comment by sdroege** [comment]
> Shall I squash commits?

That would be great, thanks! But also:

```
 error[E0364]: `client_async_tls_with_connector_and_config` is private, and cannot be re-exported
Error:    --> src/async_std.rs:152:9
    |
152 | pub use self::dummy_tls::client_async_tls_with_connector_and_config;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: consider marking `client_async_tls_with_connector_and_config` as `pub` in the imported module
   --> src/async_std.rs:152:9
    |
152 | pub use self::dummy_tls::client_async_tls_with_connector_and_config;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by sdroege** [comment]
The autobahn tests are currently know to fail. That needs some updating for API changes for which I had no time yet (can probably be taken from tokio-tungstenite).

**Comment by lambdalisue** [comment]
Squashed and force pushed.

**Comment by lambdalisue** [comment]
Thanks a lot 🎉  When are you going to release a new version which includes this?

**Comment by sdroege** [comment]
Ideally I'd like to fix the CI before that :) But if it's very urgent for you I can try making a release today or tomorrow without that.

**Comment by lambdalisue** [comment]
Thanks a lot. I really appreciate that 🥇

**Comment by sdroege** [comment]
Sorry for the delay, 0.14.0 with this is on crates.io now.

**Pull request merged**